route.go

```diff
@@ -88,6 +88,30 @@ func (r *Route) GlobalMiddleware(middlewares ...httpcontract.Middleware) {
 	r.setMiddlewares(middlewares)
 }
 
+var globalRecoverCallback func(ctx context.Context, err any)
+
+func SetGlobalRecover(callback func(ctx context.Context, err any)) {
+	globalRecoverCallback = callback
+}
+
+// the Recoverer must come first
+func (r *Route) Recover(callback func(ctx context.Context, err any)) {
+	SetGlobalRecover(callback)
+
+	r.instance.Use(func(ctx *gin.Context) {
+		defer func() {
+			if err := recover(); err != nil {
+				if globalRecoverCallback != nil {
+					globalRecoverCallback(ctx, err)
+				} else {
+					ctx.AbortWithStatusJSON(http.StatusInternalServerError, gin.H{"error": "Internal Server Error"})
+				}
+			}
```

Comment:
The logic is repeated with `middleware_timeout.go`, we can wrap it as a common method.